### PR TITLE
Fixed Docstring #3970

### LINKED ISSUE #3970: Improve Maxwell create_setup docstring, currently referring to HFSS
### Description of the feature

*** this is a demo issue for pyAEDT for Maxwell training **

Replace HFSS by Maxwell in docstring in maxwell.py function create_setup()
  

### Steps for implementing the feature

_No response_

### Useful links and references

_No response_

### PR DESCRIPTION
**Pull request by DaveTwyman**
Close #3970

### CODE REVIEW

File: pyaedt/maxwell.py [hunk 1]
```diff
@@ -1946,43 +1946,38 @@ def edit_external_circuit(self, netlist_file_path, schematic_design_name):
...
+                ----------
+
+                >>> oModule.InsertSetup
+
+                Examples
+                --------
+
+                >>> from pyaedt import Maxwell3d
+                >>> app = Maxwell3d()
+                >>> app.create_setup(setupname="Setup1", setuptype="EddyCurrent", MaximumPasses=10,PercentError=2 )
```
Comment (MaxJPRey) [inline pyaedt/maxwell.py:1980]: I would just suggest to use a different `setupname` value as `Setup1` is the default value.
```suggestion
                >>> app.create_setup(setupname="My_Setup", setuptype="EddyCurrent", MaximumPasses=10,PercentError=2 )
```

**Comment by gmalinve** [comment]
@MaxJPRey PR open for training purposes :)
I'm pretty sure someone from the team will take care of this and will fix all the issues.
Remember to fix also the SetupTemplateMaxwell.rst mentioned in the linked issue.
@IreneWoyna @DaveTwyman @tizianrot @anur7

**Comment by gmalinve** [comment]
@IreneWoyna @tizianrot @DaveTwyman could someone pick this up please? 
We need to finish editing the docstring + the SetupMaxwellTemplate.rst file in the doc.
If needed please ping me :)

**Review by gmalinve — CHANGES_REQUESTED** [review]
@IreneWoyna @tizianrot @DaveTwyman could someone pick this up please?
We need to finish editing the docstring + the SetupMaxwellTemplate.rst file in the doc.
If needed please ping me :)

**Comment by DaveTwyman** [comment]
> @IreneWoyna @tizianrot @DaveTwyman could someone pick this up please? We need to finish editing the docstring + the SetupMaxwellTemplate.rst file in the doc. If needed please ping me :)

Yes, I can finish this off, one of the checks is failing though, I'll message you

File: pyaedt/maxwell.py [hunk 2]
```diff
@@ -1946,43 +1946,38 @@ def edit_external_circuit(self, netlist_file_path, schematic_design_name):
...
+                    For more information, see :doc:`../SetupTemplatesMaxwell`.
+
+                Returns
+                -------
+                :class:`pyaedt.modules.SolveSetup.SetupMaxwell`
+                    3D Solver Setup object.
+
+                References
+                ----------
+
```
Comment (Samuelopez-ansys) [inline pyaedt/maxwell.py:1972]:
```suggestion
```

File: pyaedt/maxwell.py [hunk 3]
```diff
@@ -1946,43 +1946,38 @@ def edit_external_circuit(self, netlist_file_path, schematic_design_name):
...
+                >>> oModule.InsertSetup
+
+                Examples
+                --------
+
```
Comment (Samuelopez-ansys) [inline pyaedt/maxwell.py:1977]:
```suggestion
```

File: pyaedt/maxwell.py [hunk 4]
```diff
@@ -1946,43 +1946,38 @@ def edit_external_circuit(self, netlist_file_path, schematic_design_name):
...
-        --------
-
-        >>> from pyaedt import Maxwell3d
-        >>> app = Maxwell3d()
-        >>> app.create_setup(setupname="Setup1", setuptype="EddyCurrent", MaximumPasses=10,PercentError=2 )
+                Optional arguments are passed along with ``setuptype`` and ``setupname``.  Keyword
+                names correspond to the ``setuptype``
+                corresponding to the native AEDT API.  The list of
+                keywords here is not exhaustive.
+        ``
```
Comment (Samuelopez-ansys) [inline pyaedt/maxwell.py:1953]:
```suggestion
        
```

**Review by Samuelopez-ansys — COMMENTED** [review]
Please commit these changes

File: pyaedt/maxwell.py (hunk 4 above)
Comment (Samuelopez-ansys) [reply pyaedt/maxwell.py:1953]: In addition, all the docstrings have an extra indentation, please check others' methods, and you will see how the indentation should be. Thanks

**Comment by gmalinve** [comment]
@DaveTwyman checks fail because we didn't finish to work on the docstring and some indentations are wrong

File: pyaedt/maxwell.py [hunk 5]
```diff
@@ -1951,16 +1951,11 @@ def create_setup(self, setupname="MySetupAuto", setuptype=None, **kwargs):
...
-           This method overrides the ``Analysis.setup()`` method for the HFSS app.
-
         Parameters
         ----------
         setuptype : int, str, optional
             Type of the setup. Based on the solution type, options are
-            ``"HFSSDrivenAuto"``, ``"HFSSDrivenDefault"``,
-            ``"HFSSEigen"``, ``"HFSSTransient"`` and ``"HFSSSBR"``.
-            The default is ``"HFSSDrivenAuto"``.
+            "Magnetostatic", "EddyCurrent", Transient" etc.
```
Comment (gmalinve) [inline pyaedt/maxwell.py:1958]:
```suggestion
            "Magnetostatic", "EddyCurrent", Transient" etc.
```
Add all setup types

File: pyaedt/maxwell.py [hunk 6]
```diff
@@ -1951,16 +1951,11 @@ def create_setup(self, setupname="MySetupAuto", setuptype=None, **kwargs):
...
         setupname : str, optional
             Name of the setup. The default is ``"Setup1"``.
```
Comment (gmalinve) [inline pyaedt/maxwell.py:1960]:
```suggestion
            Name of the setup. The default is ``"Setup1"``.
```
The default name is not "Setup1"

**Review by DaveTwyman — COMMENTED** [review]
Looks ok, I'll add the missing Maxwell solver types as well

**Review by DaveTwyman — COMMENTED** [review]
Looks Good

File: pyaedt/maxwell.py [hunk 7]
```diff
@@ -1951,16 +1951,12 @@ def create_setup(self, setupname="MySetupAuto", setuptype=None, **kwargs):
...
-           This method overrides the ``Analysis.setup()`` method for the HFSS app.
-
         Parameters
         ----------
         setuptype : int, str, optional
-            Type of the setup. Based on the solution type, options are
-            ``"HFSSDrivenAuto"``, ``"HFSSDrivenDefault"``,
-            ``"HFSSEigen"``, ``"HFSSTransient"`` and ``"HFSSSBR"``.
-            The default is ``"HFSSDrivenAuto"``.
+            Type of the setup. Based on the solution type, options are:-
```
Comment (SMoraisAnsys) [inline pyaedt/maxwell.py:1957]:
```suggestion
            Type of the setup. Based on the solution type, options are:
```

Comment (SMoraisAnsys) [reply pyaedt/maxwell.py:1957]: I suppose the dash isn't needed. If it is of interest, ignore my remark (and please explain me its uses) :)

**Comment by gmalinve** [comment]
@DaveTwyman just a suggestion and based on the training sessions we had, it's better to put a meaningful comment for each commit you make. So far you have pushed all the commits with the same message maxwell.py .
It's not a problem but it's better to have a short message related to the change in your commit so it's easier for the review process :)

File: pyaedt/maxwell.py [hunk 8]
```diff
@@ -1951,16 +1951,12 @@ def create_setup(self, setupname="MySetupAuto", setuptype=None, **kwargs):
...
         Parameters
         ----------
         setuptype : int, str, optional
-            Type of the setup. Based on the solution type, options are
-            ``"HFSSDrivenAuto"``, ``"HFSSDrivenDefault"``,
-            ``"HFSSEigen"``, ``"HFSSTransient"`` and ``"HFSSSBR"``.
-            The default is ``"HFSSDrivenAuto"``.
+            Type of the setup. Based on the solution type, options are:
+            "Magnetostatic", "EddyCurrent", Transient" and
+            "Electrostatic", "DC Conduction", "Electric Transient", "AC Conduction"
```
Comment (PipKat) [inline pyaedt/maxwell.py:1959]:
```suggestion
            ``"AC Conduction"``, ``"DC Conduction"``, ```"EddyCurrent"``, ``"Electric Transient"``,
            ``"Electrostatic"``, ``"Magnetostatic"``, and ``Transient"``.
```
Options should be alphabetized and in double backticks and quotation marks.

File: pyaedt/maxwell.py [hunk 9]
```diff
@@ -1951,16 +1951,12 @@ def create_setup(self, setupname="MySetupAuto", setuptype=None, **kwargs):
...
-           This method overrides the ``Analysis.setup()`` method for the HFSS app.
-
         Parameters
         ----------
         setuptype : int, str, optional
-            Type of the setup. Based on the solution type, options are
-            ``"HFSSDrivenAuto"``, ``"HFSSDrivenDefault"``,
-            ``"HFSSEigen"``, ``"HFSSTransient"`` and ``"HFSSSBR"``.
-            The default is ``"HFSSDrivenAuto"``.
+            Type of the setup. Based on the solution type, options are:
```
Comment (PipKat) [inline pyaedt/maxwell.py:1957]:
```suggestion
            Type of the setup. Depending on the solution type, options are:
```

File: doc/source/API/SetupTemplatesMaxwell.rst [hunk 10]
```diff
@@ -8,19 +8,18 @@ You can edit a setup after it is created. Here is an example:
...
+    setup.props["MaximumPasses"] = 5
+    setup.update ()
 
 .. pprint:: pyaedt.modules.SetupTemplates.MaxwellTransient
 .. pprint:: pyaedt.modules.SetupTemplates.Magnetostatic
 .. pprint:: pyaedt.modules.SetupTemplates.Electrostatic
 .. pprint:: pyaedt.modules.SetupTemplates.EddyCurrent
 .. pprint:: pyaedt.modules.SetupTemplates.ElectricTransient
-
+.. pprint:: pyaedt.modules.SetupTemplates.ACConduction
```
Comment (gmalinve) [inline doc/source/API/SetupTemplatesMaxwell.rst:24]: @DaveTwyman please have a look at the doc build checks that fails. It says that in SetupTemplates there are no templates named "ACConduction" or "DCConduction".

**Comment by SMoraisAnsys** [comment]
> DaveTwyman

@DaveTwyman if you ever want to try to rewrite your branch history (e.g. changing commit names, squashing commits, do not hesitate to contact me)

File: doc/source/API/SetupTemplatesMaxwell.rst (hunk 10 above)
Comment (DaveTwyman) [reply doc/source/API/SetupTemplatesMaxwell.rst:24]: @gmalinve , Ok, they can be added in the future. Maybe at the same time as adjusting all the other solver templates that reference HFSS. I've removed them for now

**Review by DaveTwyman — COMMENTED** [review]
Removed reference to AC and DC Conduction templates

File: doc/source/API/SetupTemplatesMaxwell.rst (hunk 10 above)
Comment (gmalinve) [reply doc/source/API/SetupTemplatesMaxwell.rst:24]: @DaveTwyman did you remove them?

Comment (DaveTwyman) [reply doc/source/API/SetupTemplatesMaxwell.rst:24]: @gmalinve , I deleted the two lines in Pycharm and then hit commit.

**Review by DaveTwyman — COMMENTED** [review]
Removing this makes sense, I now see this is the maxwell.py file and not the rst file

File: doc/source/API/SetupTemplatesMaxwell.rst (hunk 10 above)
Comment (gmalinve) [reply doc/source/API/SetupTemplatesMaxwell.rst:24]: @DaveTwyman as you can see these lines are still here, let me try to do it on my side.

**Pull request merged**